src/Microsoft.OData.Core/UriParser/Binders/InBinder.cs

```diff
@@ -108,7 +108,8 @@ private CollectionNode GetCollectionOperandFromToken(QueryToken queryToken, IEdm
 
                     Debug.Assert(expectedType.IsCollection());
                     string expectedTypeFullName = expectedType.Definition.AsElementType().FullTypeName();
-                    if (expectedTypeFullName.Equals("Edm.String", StringComparison.Ordinal))
+
+                    if (expectedTypeFullName.Equals("Edm.String", StringComparison.Ordinal) || expectedTypeFullName.Equals("Edm.Untyped", StringComparison.Ordinal))

```

Comment:
@KenitoInc I'm assuming the dynamic property will be `Edm.Untyped`. Can you test what happens if you have a dynamic property that is not a string, e.g., an array of strings? Just to confirm it wont result into an exception being thrown...

Reply:
Do you mean `$filter=dynamicProperty in ('')`
where `dynamicProperty` is `Collection(Edm.Untyped)`?

Reply:
@KenitoInc Exactly

Reply:
Is that request valid?

Reply:
how can we know 'where dynamicProperty is Collection(Edm.Untyped)?' 

we say a dynamic property, means it's a value. Since it's a real value, then we know the type of the value. Right?
If the value type has Edm type defined in the model, we know how to serialize it.  
If the value type has not Edm type defined in the model, "actually" we don't know how to serialize it. In the latest ASP.NET Core OData, I have a default implementation by retrieving the properties of the value type (instance, public, has get) to serialize it. So, a dynamic property has no relation to "Edm.Untyped".

 Edm.Untyped or its collection is for a declared property. A declared property has two aspects. One is schema definition. The other is real property value.  We can declare a property using Edm.Untyped, or Collection(Edm.Untyped). It's from schema definition perspective. Such property can be called Untyped property. The value for an untyped property is called untyped value. Untyped value can be anything. 
"Untyped value" doesn't mean we don't know the type of value; it means we can't know the type of value until we get such value. Meanwhile, same untyped property for different entities could have different typed value.

Reply:
@KenitoInc My bad... I see your point. We're checking that the dynamic property is of `Edm.Untyped` so there's no opportunity for a `Collection(Edm.Untyped)` property should ruin the party